internal/controllers/extension_controller.go

```diff
@@ -482,19 +481,38 @@ func (r *ExtensionReconciler) resolve(ctx context.Context, extension ocv1alpha1.
 		predicates = append(predicates, catalogfilter.InMastermindsSemverRange(vr))
 	}
 
+	// Do not include versions older than currently installed unless UpgradeConstraintPolicy = 'Ignore'
+	var installedVersion string
+	if extension.Spec.Source.Package.UpgradeConstraintPolicy != ocv1alpha1.UpgradeConstraintPolicyIgnore {
```

Comment:
I'm not seeing an obvious way to reduce the nesting further without creating a `func`, which feels unnecessary just to get past the linter. What do you all think?

Reply:
I actually would suggest a separate function, I think. I'd expect you to be able to outdent the `else if` contents because you'll have the ability to return early if there is an error or the annotation doesn't exist.

Reply:
I avoided returning early here for annotation failure or semver parsing because to me this filtering felt non-critical. The `resultSet` at the end will be sorted with the highest version at the top anyway, so besides performance improvements what does removing older versions from the `resultSet` give us?

Reply:
To answer my own question a little bit, I can imagine if an installed bundle is removed from a catalog for some reason, we would want to give an error here rather than have this resolve to an older version of the bundle and cause a downgrade.